test/tasks/ods-build-typescript_test.go

```diff
@@ -113,5 +113,57 @@ func TestTaskODSBuildTypescript(t *testing.T) {
 				},
 				WantSetupFail: true,
 			},
+			"build backend typescript app": {
+				Timeout:             20 * time.Minute,
```

Comment:
Does this take 20mins?

Reply:
This might be a leftover value from when #321 wasn't implemented yet. 10 minutes should suffice, but the default of 5 minutes will not be enough due to the amount of time it takes to copy the  `node_modules`.

Reply:
Is there no way to have just one dependency in `node_modules`? We already test that a basic web server works in the other tests so maybe we can really reduce it to the bare minimum? 5 minutes is just insane ;)

Reply:
Copying the entire node_modules will presumably also copy dev dependencies. 
If there would be a command to copy only the production node_modules, that might be better.  
An alternative I have seen in our build is to have another production build done. 
One could perhaps copy package.json and package-lock.json if existing and then rely on the Docker build to do an `npm i --only=production` or `npm ci --only=production`. 
Folks with more node experience, please correct me if this is nonsense!

Reply:
If my previous comment made sense this could also fix issue #357 

One concern is that the production build may not actually run as we did not test it, for example if a true dependency was moved to the devDependencies. Not sure whether this would typically be caught by typescript.

Reply:
One additional note is that the go build does the production build during the build task after running the tests which essentially does a dev build. 
The files from the production build (`go build`) are then copied to the docker context.

We could presumably do the same for node.js although

Reply:
@henrjk Hehe, true.

However I think the Go and TypeScript situation is different:

* Go modules has no way to distinguish between dev/test dependencies and production dependencies. In all cases, only the dependencies needed are included in the produced binary.
* `go test` actually compiles a test binary. The only difference to `go build` is that it includes `_test.go` files and (in our case) is executed with slightly different args (that should not affect anything).

I think both properties should allow to neglect this issue (plus we have a statically compiled result and the Kubernetes startup probe). That said, even if we were to reverse the order of `go build` and `go test`, nothing would change - the tests would still run against the test binary produced by `go test`.

Contrast this to the TypeScript case where there is a distinction between dev and prod dependencies, which should result in a very different result. Unfortunately I do not know enough about the situation to know what issues could result there.

Reply:
@michaelsauter  Right! 
go compile builds will result in all production dependencies be in the final binary while go test does the same but also includes the test dependencies.
So in go we actually have the tools keep these separate by design and it would not matter if we switched the order. Also a go build would catch all missing dependencies if this can even happen - considering go test would have already passed.

Now if we wanted to mirror this we would have to have two different environments e.g dirs one for testing similar to how it is now and then a second `deploy` environment to build and potentially test the deployed code.

At the moment I am leaning towards having the production build happen in the docker build and thus only copying `package*.json` would be sufficient. Perhaps there could be variable to control whether to copy the entire node_modules folder.

Reply:
Does it make sense to fix this in #357 and continue the discussion there? What do you think?

Reply:
@henninggross Agree. Let's just deal with the overly long timeout here.

Reply:
I am unresolving so it is visible when linking from #357.